ntoskrnl/ex/locale.c

```diff
@@ -293,36 +550,41 @@ NtSetDefaultLocale(IN BOOLEAN UserProfile,
     }
     else
     {
-        /* Otherwise, open the key */
-        Status = ZwOpenKey(&KeyHandle, KEY_SET_VALUE, &ObjectAttributes);
+        /* We have a locale, validate it */
+        Status = ExpValidateNlsLocaleId(DefaultLocaleId);
         if (NT_SUCCESS(Status))
```

Comment:
You may save some indentation levels by using `if (!NT_SUCCESS(Status)) goto Cleanup;` construct, for here and for the ZwOpenKey call below.

Reply:
I don't think having extra indentation hurts anybody, unless we are talking about a bloat load of more indentation spaces.